src/TwigComponent/src/ComponentFactory.php

```diff
@@ -184,7 +184,8 @@ private function preMount(object $component, array $data, ComponentMetadata $com
             $newData = $component->{$method->name}($data);
 
             if (null !== $newData) {
-                $data = $newData;
+                // merge data/options so that extra data is not lost between hooks
+                $data = $newData + $data;
```

Comment:
Hmm i'm not sure here .... as you cannot "transform" data with preMount event there. 

That would be a big BC break :/ 

I would really prefer we only update documentation for those who want to uses partial validation with OptionResolver if we can.

Reply:
> as you cannot "transform" data with preMount event there

What do you mean by that? Example?

Reply:
Let's say you call a component like that

<twig:Alert level=warning />

In your preMount method, you transform "level" into "class", and so you remove level

If you merge newData and data, will "level" be dropped.. or kept as an attribute ?

Reply:
Yeah you right, removed data will be back due to the base array concat..
If it is a feature, we maybe need a test to keep that.